src/transaction/log_recovery_redo_parallel.cpp

```diff
@@ -500,15 +99,67 @@ namespace cublog
       void log_perf_stats () const;
       void accumulate_perf_stats (cubperf::stat_value *a_output_stats, std::size_t a_output_stats_size) const;
 
+      bool is_idle () const;
+
+      inline void push_job (redo_parallel::redo_job_base *a_job);
+
+      inline log_lsa get_not_applied_log_lsa ();
+
+    private:
+      inline void pop_jobs (std::unique_ptr<redo_job_vector_t> &a_out_job_vec);
+
+      inline void set_not_applied_log_lsa_from_push_side (const log_lsa &a_log_lsa);
+      inline void set_not_applied_log_lsa_from_pop_side (const log_lsa &a_log_lsa);
+
     private:
       const std::size_t m_task_idx;
+      const bool m_do_monitor_not_applied_log_lsa;
+
       redo_parallel::task_active_state_bookkeeping &m_task_state_bookkeeping;
-      redo_parallel::redo_job_queue &m_queue;
 
       cubperf::statset_definition m_perf_stats_definition;
       perf_stats m_perf_stats;
 
       log_rv_redo_context m_redo_context;
+
+      std::unique_ptr<redo_job_vector_t> m_produce_vec;
```

Comment:
Why unique pointer to vector?
```suggestion
      redo_job_vector_t m_produce_vec;
```

Reply:
Evolution: raw pointer -> unique pointer -> vector.